cf_xarray/tests/test_accessor.py

```diff
@@ -1237,6 +1237,66 @@ def test_cmip6_attrs():
     assert da.cf.axes["Y"] == ["nlat"]
 
 
+def test_custom_criteria():
+    my_custom_criteria = {
+        "ssh": {
+            "standard_name": {"regex": ("sea_surface_elev*",)},
```

Comment:
Sorry to keep flip-flopping on this but another way would be

``` python
ssh = {"standard_name": "sea_surface_elev*|sea_surface_height"}
```

I think this contains both cases pretty nicely. We just expect the users to always provide regex-like input if they want multiple options for "standard name" (i.e. with `|`)

What do you think?

Reply:
Yes now that you say it, this is simpler rather than building in the structure. But, if the user wants an exact match we still need a separate case for that don't we?

Reply:
I'll experiment with this — I suppose the user could have to use "$" at the end of the string to make an exact match with regex.

Reply:
The new version is much simpler — I like it a lot more. It will be easy with enough examples to demo.